.github/workflows/reusable-build.yml

```diff
@@ -120,27 +123,39 @@ jobs:
           # DEBUG: get character count of key
           wc -c certs/private_key.priv
 
-      - name: Get current version
-        id: labels
+      - name: Pull build image
         uses: Wandalen/wretry.action@v2.1.0
         with:
           attempt_limit: 3
           attempt_delay: 15000
           command: |
-            set -eo pipefail
-            skopeo inspect docker://quay.io/${{ env.SOURCE_ORG }}/${{ env.SOURCE_IMAGE }}:${{ matrix.fedora_version }} > inspect.json
-            ver=$(jq -r '.Labels["org.opencontainers.image.version"]' inspect.json)
-            linux=$(jq -r '.Labels["ostree.linux"]' inspect.json)
-            if [ -z "$ver" ] || [ "null" = "$ver" ]; then
-              echo "inspected image version must not be empty or null"
-              exit 1
-            fi
-            if [ -z "$linux" ] || [ "null" = "$linux" ]; then
-              echo "inspected image linux version must not be empty or null"
-              exit 1
-            fi
-            echo "SOURCE_IMAGE_VERSION=$ver" >> $GITHUB_ENV
-            echo "SOURCE_IMAGE_LINUX=$linux" >> $GITHUB_ENV
+            # pull the base image used for FROM in containerfile so
+            # we can retry on that unfortunately common failure case
+            podman pull ${{ env.FQ_SOURCE_IMAGE }}
+
+      - name: Get current version
+        run: |
+          set -eo pipefail
+
+          # Get kernel version with podman
+          container_name="fq-$(uuidgen)"
+          podman run --entrypoint /bin/bash --name "$container_name" -dt "${{ env.FQ_SOURCE_IMAGE }}"
+          installed_packages=$(podman exec "$container_name" rpm -qa)
```

Comment:
this is pretty much what I was expecting to do...

except, i think run `podman exec "$container_name" rpm-ostree install dnf`
and also to add the appropriate repo for asus, fsync or surface kernel flavors...
which then will allow `podman exec "$container_name" dnf search kernel` and you can get the "would be installed version" (in case of surface its' `kernel-surface` not `kernel`).